Framework/Core/Windows/Devices/I2c/I2cDevice.cs

```diff
@@ -0,0 +1,206 @@
+using System;
+using Microsoft.SPOT.Hardware;
+
+namespace Windows.Devices.I2c
+{
+    public struct I2cTransferResult
+    {
+        public I2cTransferStatus Status;
+        public uint BytesTransferred;
+    }
+
+    public sealed class I2cDevice : IDisposable
+    {
+        // Construction and destruction
+
+        internal I2cDevice(ushort slaveAddress, I2cBusSpeed busSpeed)
+        {
+            int clockRateKhz = 100;
+            if (busSpeed == I2cBusSpeed.FastMode)
+            {
+                clockRateKhz = 400;
+            }
+
+            m_configuration = new I2CDevice.Configuration(slaveAddress, clockRateKhz);
+
+            lock (s_deviceLock)
+            {
+                if (s_device == null)
+                {
+                    s_device = new I2CDevice(m_configuration);
+                }
+
+                ++s_deviceRefs;
+            }
+        }
+
+        ~I2cDevice()
+        {
+            Dispose(false);
+        }
+
+        // Public methods
+
+        public static string GetDeviceSelector()
+        {
+            return Enumeration.DeviceInformation.s_I2cPrefix;
+        }
+
+        public static string GetDeviceSelector(string friendlyName)
+        {
+            return friendlyName;
+        }
+
+        // FUTURE: This should be "Task<I2cDevice> FromIdAsync(...)"
+        public static I2cDevice FromId(string deviceId, I2cConnectionSettings settings)
+        {
+            if ((settings.SlaveAddress > 0) || (settings.SlaveAddress < 127))
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            if (deviceId != "I2C0")
+            {
+                // TODO: Is this the right exception?

```

Comment:
Make a decision on the exception and go with it, don't leave this commented out as it just ignores this known bad condition.